### PR TITLE
[WIP] [Feature] Implemented `is_optional` feature

### PR DESCRIPTION
**Pull request by dogacanozil**

### CODE REVIEW

File: src/dtflw/input_table.py [hunk 1]
```diff
@@ -40,4 +61,4 @@ def validate(self, strict: bool = False):
             No used here but added for compatiblity with output_table.validate.
         """
         if self.needs_eval():
-            raise Exception("Required input not found.")
+            raise Exception(f"Required input not found.")
```
Comment (dr-BEat) [inline src/dtflw/input_table.py:64]: Why the format here?

Comment (dogacanozil) [reply src/dtflw/input_table.py]: @dr-BEat Added mistakenly I guess the PR is not fully ready yet, I will keep pushing changes.

File: src/dtflw/lazy_notebook.py [hunk 2]
```diff
@@ -82,14 +85,26 @@ def input(self, name: str, file_path: str = None, source_table: str = None) -> L
                 self.rel_path
             )
 
-        elif not self.ctx.storage.is_abs_path(input_file_path):
+        elif not self.ctx.storage.is_abs_path(input_file_path) and not input_file_path=="Input file is optional and does not exist!":
```
Comment (User1000) [inline src/dtflw/lazy_notebook.py:88]: Let's go with just an empty string rather than specific string value

File: CHANGES.md [hunk 3]
```diff
@@ -1,6 +1,10 @@
 # Changes
 _All notable changes to the codebase are documented in this file._
 
+## [0.6.7] - 26.09.2024
+- Added `is_optional` parameter for `LazyNotebook.inputs` and `input_table` in order to enable `input_table`
+to be optional and not throwing an expception with missing input table in the specified `file_path`
+
```
Comment (User1000) [inline CHANGES.md:7]: Added `is_optional` parameter to the `LazyNotebook.inputs` and `input_table`. This enhancement allows to make the input_table optional, preventing it from raising an exception when the specified `file_path` is missing.

File: src/dtflw/input_table.py [hunk 4]
```diff
@@ -21,10 +22,27 @@ def abs_file_path(self) -> str:
...
+    def is_optional(self) -> bool:
+        """
+        Returns the optionality indicator of the table.
+        """
+        return self.__is_optional
 
     def needs_eval(self) -> bool:
         """
-        Returns True if the table needs to be evaluated, and False otherwise.
+        Returns False, if the file path is sepcifically indicated as "Input file is optional and does not exist!",
```
Comment (User1000) [inline src/dtflw/input_table.py:35]: Correct doc string, please. Now it doesn't rely on "Input file is optional and does not exist!" value.

File: src/dtflw/lazy_notebook.py [hunk 5]
```diff
@@ -66,6 +66,9 @@ def input(self, name: str, file_path: str = None, source_table: str = None) -> L
         source_table: str
             Name of the source table to pass an an input. 
             If 'source_table' is not given then 'name' is used.
+        is_optional: bool
+            Indicator if the input table is optional or not. 
+            If is_optional == True then input table is not necessarily be exist in the file_path.
```
Comment (User1000) [inline src/dtflw/lazy_notebook.py:71]: If is_optional == True, then the input table may not necessarily exist in the specified file_path.

File: src/dtflw/lazy_notebook.py [hunk 6]
```diff
@@ -82,14 +85,25 @@ def input(self, name: str, file_path: str = None, source_table: str = None) -> L
                 self.rel_path
             )
 
-        elif not self.ctx.storage.is_abs_path(input_file_path):
+        elif not self.ctx.storage.is_abs_path(input_file_path) and not input_file_path=="":
```
Comment (User1000) [inline src/dtflw/lazy_notebook.py:88]: Please add spaces: and not input_file_path == ""

Comment (User1000) [reply src/dtflw/lazy_notebook.py:88]: How about `and input_file_path != ""` ?

File: src/dtflw/lazy_notebook.py [hunk 7]
```diff
@@ -82,14 +85,25 @@ def input(self, name: str, file_path: str = None, source_table: str = None) -> L
...
             # Bind the input to a given specific file
             input_file_path = self.ctx.storage.get_abs_path(input_file_path)
 
-        self.__inputs[name] = InputTable(
+        input_table = InputTable(
```
Comment (User1000) [inline src/dtflw/lazy_notebook.py:92]: let's combine these two statements into one. We can calc input_file_path and then create InputTable object

Comment (dogacanozil) [reply src/dtflw/lazy_notebook.py:95]: @User1000 Done!

File: src/dtflw/lazy_notebook.py (hunk 6 above)
Comment (dogacanozil) [reply src/dtflw/lazy_notebook.py]: @User1000 Done!

File: src/dtflw/lazy_notebook.py (hunk 5 above)
Comment (dogacanozil) [reply src/dtflw/lazy_notebook.py]: @User1000 Done!

File: src/dtflw/input_table.py (hunk 4 above)
Comment (dogacanozil) [reply src/dtflw/input_table.py]: @User1000 Done!

File: CHANGES.md (hunk 3 above)
Comment (dogacanozil) [reply CHANGES.md:6]: @User1000 Done!

File: src/dtflw/lazy_notebook.py (hunk 2 above)
Comment (dogacanozil) [reply src/dtflw/lazy_notebook.py]: @User1000 Done!

**Review by dogacanozil — COMMENTED** [review]
@User1000  I have checked the pipeline and it is working fine as aexpected

**Review by User1000 — APPROVED** [review]

**Pull request merged**